stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -294,6 +294,22 @@ end
     return A
 end
 
+@propagate_inbounds function fill!(A::UpperTriangular, x)
+    iszero(x) || throw(ArgumentError("cannot set index in the lower triangular part " *
+            "($i, $j) of an UpperTriangular matrix to a nonzero value ($x)"))
```

Comment:
Variables `i` and `j` don't seem to be set anywhere?

Reply:
Ah, good catch! I had just copied it from `setindex!`. Should have checked closely.

Reply:
Updated now